### PR TITLE
build: add CI for OpenBSD and FreeBSD

### LINKED ISSUE #3510: Add OpenBSD/FreeBSD to CI
> Regarding OpenBSD CI - is that something we could incorporate into the GitHub Actions CI?

After a little searching, I found https://github.com/cross-platform-actions/action which looks promising.

_Originally posted by @jamessan in https://github.com/libuv/libuv/issues/3452#issuecomment-1059280887_

### PR DESCRIPTION
**Pull request by jamessan**
There are some existing test failures on both OSes, and I've adjusted some of the asserts so it's easier to see why they're failing.

I'll try to find some time to look into the failures, but I thought it would be better to have this up for discussion rather than delay it due to those.

Closes #3510

### CODE REVIEW

**Comment by bnoordhuis** [comment]
@jamessan do you still want to pursue this?

**Comment by jamessan** [comment]
Someone more familiar with libuv internals and BSD should take it, or skip the failing tests to establish a baseline.

I don't currently have the time to understand what the tests are trying to validate and why they're failing.  I was hoping to spark some interest in having some BSD CI.

If nothing else, at least the extra diagnostics in the tests should be helpful. :⁠-⁠)

**Comment by santigimeno** [comment]
Can we have those jobs added even if we have currently some failing tests?

**Comment by bnoordhuis** [comment]
We could skip failing tests if there aren't too many. I'll rebase and take stock.

**Comment by bnoordhuis** [comment]
freebsd: `Terminating VM` (during build)

openbsd: `/bin/ksh: <stdin>[4]: cmake: not found`

Guess that means it's not ready for inclusion in the CI matrix yet.

@jamessan if you can get it to a point where the test runner runs to completion, I can take it from there. Otherwise, I suppose we should close this.

**Comment by jamessan** [comment]
@bnoordhuis The FreeBSD error was further back in the log:

```
/Users/runner/work/libuv/libuv/src/unix/thread.c:389:9: warning: implicit declaration of function 'sched_getcpu' [-Wimplicit-function-declaration]
  cpu = sched_getcpu();
        ^
1 warning generated.
...
ld: error: undefined symbol: sched_getcpu
>>> referenced by thread.c
>>>               thread.c.o:(uv_thread_getcpu) in archive libuv_a.a
cc: error: linker command failed with exit code 1 (use -v to see invocation)
```

Looks like that was introduced by https://github.com/libuv/libuv/pull/3803.  As far as I can tell, `sched_getcpu()` is glibc-specific, so it looks like the checks for when to use it are wrong.

**Comment by santigimeno** [comment]
> Looks like that was introduced by #3803. As far as I can tell, `sched_getcpu()` is glibc-specific

I think it was added in FreeBSD 13.1. So this check should be added: 
```
__FreeBSD_version >= 1301000
```

**Comment by jamessan** [comment]
Ok, @bnoordhuis.  I had to make some changes to the thread affinity support to get libuv to build/run tests on the BSDs.  The remaining test failures appear to be back to what I was seeing when I first opened the PR.

**Comment by vtjnash** [comment]
@jamessan could you mark the rest as TEST_SKIP, so we can merge your current work before it gets stale?

File: .github/workflows/CI-unix.yml [hunk 1]
```diff
@@ -126,3 +126,36 @@ jobs:
...
+    name: build-${{ matrix.config.os }}-${{ matrix.config.version }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          # The OS versions supported are specific to the version of the action
+          # https://github.com/cross-platform-actions/action/blob/master/changelog.md
+          - { os: freebsd, version: '13.1', runner: macos-11, install: 'pkg install -y' }
+          - { os: openbsd, version: '7.2',  runner: macos-11, install: 'pkg_add -I' }
```
Comment (vtjnash) [inline .github/workflows/CI-unix.yml:142]: Can we use ubuntu-latest here? There is generally less availability of macos runners. The cross-platform-actions seems to imply it is operational

Comment (jamessan) [reply .github/workflows/CI-unix.yml:142]: The ubuntu runners use qemu, which I figured would be slower than the macOS hypervisor.  I'll give it a shot and see.

Comment (jamessan) [reply .github/workflows/CI-unix.yml]: FreeBSD seems fine with the ubuntu runner, but OpenBSD was hitting the 30m timeout when it passes in ~10 minutes on macos-11 runner.

**Review by vtjnash — APPROVED** [review]

**Review by bnoordhuis — APPROVED** [review]

**Comment by bnoordhuis** [comment]
I'm considering dropping the freebsd and openbsd buildbots again. They're disproportionally flaky compared to other platforms and no one seems invested enough to investigate so they just end up being a nuisance and a time sink.

**Pull request merged**